lib/node_modules/@stdlib/math/base/special/roundsd/src/main.c

```diff
@@ -0,0 +1,102 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+
+// MODULES //
+
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/special/pow.h"
+#include "stdlib/math/base/assert/is_infinite.h"
+#include "stdlib/math/base/special/log10.h"
+#include "stdlib/math/base/special/ln.h"
+#include "stdlib/math/base/special/abs.h"
+#include "stdlib/math/base/special/floor.h"
+#include "stdlib/number/float64/base/exponent.h"
+#include "stdlib/math/base/special/round.h"
+#include "stdlib.h"
+
+
+// MAIN //
+
+/**
+* Rounds a numeric value to the nearest number with \\(n\\) significant figures.
+*
+* @param x - input value
+* @param n - number of significant figures
+* @param b=10 - base
+* @returns {number} rounded value
+*
+* @example
+* double v = roundsd( 3.141592653589793, 3, 10 );
+* // returns 3.14
+*
+* @example
+* double v = roundsd( 3.141592653589793, 1, 10 );
+* // returns 3.0
+*
+* @example
+* double v = roundsd( 12368.0, 2, 10 );
+* // returns 12000.0
+*
+* @example
+* double v = roundsd( 0.0313, 2, 2 );
+* // returns 0.03125
+*/
+double stdlib_base_roundsd( const double x, const int32_t n, const int32_t b ) {
+	double exp;
+	double s;
+	double y;
+	if (
+		stdlib_base_is_nan( x ) ||
+	    stdlib_base_is_nan( n ) ||
+		n < 1 ||
+		stdlib_base_is_infinite( n )
+	) {
+		return 0.0/0.0; //NaN
+	}
```

Comment:
```suggestion
	if ( stdlib_base_is_nan( x ) || n < 1 ) {
		return 0.0/0.0; //NaN
	}
```

No `NaN` or infinity checks for integer values.